### DISCUSSION TITLE
Self hosting: oAuth

### DISCUSSION THREAD

**Discussion by burnoutprojects** [supabase/supabase#3895] [Questions] [answered]
Hello, how can i use oAuth on self hosting instance on supabase? I read the docs but i don't found anything related to that concept. Also i look the docker compose and env files and i don't find any solution to that problem.

Any suggestions?

**Comment by awalias** [comment, marked as answer]
you just need to add these env vars for a given provider: https://github.com/supabase/gotrue#external-authentication-providers

(make sure not to miss the `EXTERNAL_X_REDIRECT_URI`)

for getting your client secrets etc you can follow the platform guides: https://supabase.io/docs/guides/auth/auth-gitlab

**Reply by burnoutprojects** [reply]
Thanks for the answer, i check it later today

**Reply by Auxx** [reply]
@awalias how do I set `GOTRUE_EXTERNAL_X_Y` variables? Tried adding `GOTRUE_EXTERNAL_GITHUB_Y` example from the docs to the .env file and nothing changed. Tried adding it to `docker-compose.yml` and nothing again.
